aerogear/mobile-security-service
# (Aerogear 8913) - Rest API changes to allow GET and POST App

## Motivation
https://issues.jboss.org/browse/AEROGEAR-8913

## What
- Create POST endpoint to create app
- Impl QueryParameter into GET endpoint to return the app by appID
- Create PATCH endpoint to update app

## Why
- To allow GET and create and update apps using the API
- Attend MDC/Operator requirements

## How
- Create new POST endpoint "/api/apps""
- Add QueryParam "?appId" to the /api/apps GET endpoint

## Verification Steps
- [ ] **Check if CI finished with success**
- [ ] **Create an app from the POST endpoint**
* Call the endpoint `localhost:3000/api/apps` with the following body.
```json
{"appId": "com.aerogear.testpost", "appName": "Test Post" }
```
<img width="1093" alt="Screenshot 2019-04-02 at 16 41 56" src="https://user-images.githubusercontent.com/7708031/55416339-84875e80-5566-11e9-953e-36ebdf00a47c.png">

* Check that the app was created with success
<img width="948" alt="Screenshot 2019-04-02 at 16 41 32" src="https://user-images.githubusercontent.com/7708031/55416345-881ae580-5566-11e9-9c87-6e7cc59840e6.png">

- [ ] **Update the name of an app already created by the POST endpoint** 
* Call the endpoint `localhost:3000/api/apps` with the following body.
```json
{"appId": "com.aerogear.testpost", "appName": "Test Post New Name" }
```
<img width="1136" alt="Screenshot 2019-04-02 at 16 42 09" src="https://user-images.githubusercontent.com/7708031/55416273-63267280-5566-11e9-83ce-0d43e1cd5928.png">

* Check that the app was updated with success
<img width="514" alt="Screenshot 2019-04-02 at 16 42 25" src="https://user-images.githubusercontent.com/7708031/55416261-5d309180-5566-11e9-873c-6cc5c8fff5da.png">

- [ ] **Check to GET the app by AppID**
* Call the the GET endpoint with the Query Param `localhost:3000/api/apps?appId=com.aerogear.testpost`
<img width="684" alt="Screenshot 2019-04-02 at 16 45 03" src="https://user-images.githubusercontent.com/7708031/55416533-daf49d00-5566-11e9-9920-6d01ba0fa619.png">

- [ ] **re-binding the app - Update the deleted_at with NULL when the post is called**
* Set the current data on the field deleted_at of the app

```sql
UPDATE app
SET deleted_at = NOW()
WHERE app_id = 'com.aerogear.testpost'
```

* Call the POST endpoint for the app `localhost:3000/api/apps` with the appId in the body.
<img width="1101" alt="Screenshot 2019-04-02 at 16 55 16" src="https://user-images.githubusercontent.com/7708031/55417210-252a4e00-5568-11e9-9550-32361fff45c6.png">

* Check that the deleted_at is NULL now. 
<img width="974" alt="Screenshot 2019-04-02 at 16 55 30" src="https://user-images.githubusercontent.com/7708031/55417221-28253e80-5568-11e9-89e8-1e6df93ac12f.png">


## Checklist:


## Progress


## Aditional
* Some changes were requested
* The name update is now in a PATCH endpoint for the apps.
